File: bonus/go/poll/Help.go
package poll

import (
	"fmt"
	"os"
)

func PrintHelp(pgrm string, err *Error) {
	if err != &HelpError {
		fmt.Println(err.Message)
	}
	fmt.Println("Usage")
	fmt.Printf("\t%s pSize sSize p\n\n", pgrm)
	fmt.Println("DESCRIPTION:")
	fmt.Println("\tpSize\tsize of the population")
	fmt.Println("\tsSize\tsize of the sample (supposed to be representative)")
	fmt.Println("\tp\tpercentage of voting intentions for a specific candidate")
	os.Exit(err.Code)
}

File: bonus/haskell/src/Help.hs
module Help
    ( help
    ) where

help :: String -> IO ()
help pgrm =
    putStrLn "Usage\n" >> putStrLn ("\t" ++ pgrm ++ " pSize sSize p\n\n") >> putStrLn "DESCRIPTION:\n" >>
    putStrLn "\tpSize\tsize of the population\n" >>
    putStrLn "\tsSize\tsize of the sample (supposed to be representative)\n" >>
    putStrLn "\tp\tpercentage of voting intentions for a specific candidate"
